hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/ContainerEndpoint.java

```diff
@@ -141,10 +140,18 @@ public Response getContainers(
       // Send back an empty response
       return Response.status(Response.Status.NOT_ACCEPTABLE).build();
     }
+    if (prevKey > 0) {
+      // Increase the limit by 1 to fetch one additional container
+      // since we are excluding the container with the same ID as prevKey
+      limit = limit + 1;
```

Comment:
We can avoid increasing limit, as the getContainers() return results in sorted order, instead we can increase prevKey by 1 here as anyway we want to discard prevKey container info.

Reply:
My Initial thought for raising the limit was that we might encounter non-sequential container ID's 

When we sort container IDs in ascending order, we expect to have a sequence of consecutive numbers. For example 1, 2, 3, 4, 5, 6 would represent a proper sequence where each number follows the previous one. However, it's important to note that even though we sort the container IDs, it doesn't guarantee a fixed sequence.

In some cases, we may encounter container IDs that are not in a consecutive or sequential order. For instance, we might have container IDs like 1, 3, 4, 10, 23, 99, where there are gaps or missing numbers between the IDs. This can occur due to various reasons such as containers being deleted or missing. These missing or deleted containers result in gaps within the sequence of container IDs.

But after checking out the code in the [getContainers()](https://github.com/apache/ozone/blob/7d1aaef989f3749014c870000c84a6645d6fb43b/hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ContainerManagerImpl.java#LL144C1-L146C1) it turns out even if we provide a containerId that does not exist it will still filter them out and start with a container ID that exists. So I believe the code changees suggested by @ashishkumar50 will work out fine.

Reply:
I have ran the existing UT's for the new changes and they work out fine !!